sdk/Pulumi/Resources/ComponentResource.cs

```diff
@@ -52,6 +53,49 @@ public ComponentResource(string type, string name, ComponentResourceOptions? opt
             this.remote = remote;
         }
 
+        internal static Dictionary<string, object?> CollectOutputProperties(ComponentResource instance)
+        {
+            var outputs = new Dictionary<string, object?>();
+            var currentType = instance.GetType();
+            foreach (var prop in currentType.GetProperties())
+            {
+                if (prop.Name == nameof(Urn))
+                {
+                    continue;
+                }
+
+                var outputAttribute =
+                    prop.GetCustomAttributes(typeof(OutputAttribute), false)
+                        .FirstOrDefault();
+                if (outputAttribute is OutputAttribute attr && attr.Name != null)
+                {
+                    // when using [Output("<name>")] we will export the value of this property
+                    // named as the provided <name>
+                    var value = prop.GetValue(instance);
+                    outputs.Add(attr.Name, value);
+                }
+                else if (outputAttribute is OutputAttribute)
+                {
+                    // otherwise of we only have [Output] we will simply use the name of the property itself
+                    // when exporting the value
+                    var value = prop.GetValue(instance);
+                    outputs.Add(prop.Name, value);
+                }
+                else
+                {
+                    var propertyType = prop.PropertyType;
+                    // only if the type is Output<T> then we will consider it an output property
```

Comment:
I'm inclined to not include these. Be specific that it's only things tagged [OutputAttribute] that get auto exported, a component is free to have non-Output\<T> things that are still "outputs", I don't think we should privilege the Output\<T> type.

Reply:
The reasoning here if a user has specified a property as `Output<T>` then it feels redundant to also mark it as `[Output]`. 

> a component is free to have non-Output<T> things that are still "outputs"

Is there any use for this? why have a field that is only local to the component 🤔

Reply:
You might have a component that calculates a non-output result based on it's inputs. That result might be passed down to child resources but also returned to the user to be used later as well. So it's not "only local to the component" its still a public property, its just not Output-y.

Reply:
Alright that makes sense ✅ I've updated the function and now only properties marked with `[Output]` are considered outputs of the component to be registered